pyproject.toml

```diff
@@ -47,7 +47,9 @@ dependencies = [
   "lmfit",
   "scipy",
   "numpy",
+  "orjson",
   "scipp",
+  "typing-extensions",
```

Comment:
i don't know if it's worth a comment next to this saying why this is required - we'll have to update it every time we want to go up a version of python..?

Reply:
was thinking we'd need to change the comment depending on if the new version has the feature we're relying on typing-extensions for. 

Do we actually need it though - it looks like we're using it for TypeVar?

Reply:
yes we do for `default`

Reply:
ok - ignore me, sorry, i think im confusing things

Reply:
We're using it for `TypeVar` that allows a `default`, yes. That's a python 3.13+ feature otherwise.

The reason I used it is so that we can still take arguments of type `SimpleDae` (which allows any reducer/waiter/controller), but _also_ explicitly specify which ones are wanted for plans where that makes sense

Reply:
makes sense - i was just wondering whether it's worth adding that as a comment next to this line, but maybe that'd be stupid - we don't just justify other dependencies

Reply:
Not stupid - just added justifications for all of our main deps.